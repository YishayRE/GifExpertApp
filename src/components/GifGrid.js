//?lo comentado es sin usar un custom Hook
import React, { Fragment} from 'react'
//import React, { Fragment, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
//import { getGifs } from '../helpers/getGifs';

const GifGrid = ({category}) => {
    /*
    const [Images, setImages] = useState([]);

    useEffect(() => {
        getGifs(category)
            .then(setImages);
    }, [category]);
*/
    //getGifs();

    const {data, loading} = useFetchGifs(category);
    
    return (
        <Fragment>
            <h3 className="animate__animated animate__bounce">{category}</h3>
            
            {/*loading ? 'Cargando...' : 'Data Cargada'*/}
            {loading && <p>Loading</p>}

            <div className="card-grid">
                {
                    data.map(img =>
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    )
                }
            </div>
            
            {/*
            <div className="card-grid">
                {
                    Images.map(img =>
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    )
                }
            </div>
            */}
        </Fragment>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
