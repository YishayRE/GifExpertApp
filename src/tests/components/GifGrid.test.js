import '@testing-library/jest-dom';
import React from "react";
import { shallow } from "enzyme";

import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

//?para mejor visualizaciÃ³n, hacer cada modificacion dentro de 
//?cada test
describe('Pruebas en el componente GifGrid', () => {
    const category = 'Hola';
    const gifs = [{
        id:'ABC',
        url:'url.com',
        title: 'Hola'
    }];

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: true
    });

    let wrapper = shallow(<GifGrid category={category}/>);

    test('Debe de mostrar <GifGrid/>', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar items cuando cargan las imagenes de useFetchGifs', () => {
        expect(wrapper).toMatchSnapshot();
    });
});