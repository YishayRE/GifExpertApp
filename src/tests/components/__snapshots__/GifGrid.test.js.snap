// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de mostrar <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Hola
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Hola"
      url="url.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe de mostrar items cuando cargan las imagenes de useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Hola
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Hola"
      url="url.com"
    />
  </div>
</Fragment>
`;
